server/concept/ThingService.java

```diff
@@ -198,35 +192,97 @@ private void getPlayersByRoleType(Relation relation, UUID reqID) {
+        Attribute attribute = getThing(thingReq).asAttribute();
+        ConceptProto.Attribute.GetOwners.Req getOwnersReq = thingReq.getAttributeGetOwnersReq();
+        FunctionalIterator<? extends Thing> things;
+        if (getOwnersReq.hasThingType()) things = attribute.getOwners(getThingType(getOwnersReq.getThingType()));
+        else things = attribute.getOwners();
+        transactionSvc.stream(things, reqID, owners -> getOwnersResPart(reqID, owners));
     }
 
-    private void removePlayer(Relation relation, ConceptProto.Relation.RemovePlayer.Req removePlayerReq, UUID reqID) {
-        relation.removePlayer(getRoleType(removePlayerReq.getRoleType()), getThing(removePlayerReq.getPlayer()).asThing());
-        transactionSvc.respond(removePlayerRes(reqID));
+    private void requestNotSet(ConceptProto.Thing.Req thingReq, UUID uuid) {
+        throw TypeDBException.of(UNKNOWN_REQUEST_TYPE);
     }
 
-    private void getOwners(Attribute attribute, ConceptProto.Attribute.GetOwners.Req getOwnersReq, UUID reqID) {
-        FunctionalIterator<? extends Thing> things;
-        switch (getOwnersReq.getFilterCase()) {
-            case THING_TYPE:
-                things = attribute.getOwners(getThingType(getOwnersReq.getThingType()).asThingType());
-                break;
-            case FILTER_NOT_SET:
+    private static <T extends Concept> T notNull(@Nullable T concept) {
+        if (concept == null) throw TypeDBException.of(MISSING_CONCEPT);
+        return concept;
+    }
+
+    private ThingType getThingType(ConceptProto.ThingType protoThingType) {
+        switch (protoThingType.getTypeCase()) {
+            case ENTITY_TYPE: return getEntityType(protoThingType.getEntityType()).asThingType();
+            case RELATION_TYPE: return getRelationType(protoThingType.getRelationType()).asThingType();
+            case ATTRIBUTE_TYPE: return getAttributeType(protoThingType.getAttributeType()).asThingType();
+            case TYPE_NOT_SET:
             default:
-                things = attribute.getOwners();
+                throw TypeDBException.of(BAD_VALUE_TYPE, protoThingType.getTypeCase());
         }
+    }
 
-        transactionSvc.stream(things, reqID, owners -> getOwnersResPart(reqID, owners));
+    private EntityType getEntityType(ConceptProto.EntityType protoType) {
+        return conceptMgr.getEntityType(protoType.getLabel());
     }
 
+    private RelationType getRelationType(ConceptProto.RelationType protoType) {
+        return conceptMgr.getRelationType(protoType.getLabel());
+    }
+
+    private AttributeType getAttributeType(ConceptProto.AttributeType protoType) {
+        return conceptMgr.getAttributeType(protoType.getLabel());
+    }
+
+    private RoleType getRoleType(ConceptProto.RoleType protoRoleType) {
+        RelationType relationType = conceptMgr.getRelationType(protoRoleType.getScope());
+        if (relationType != null) return relationType.getRelates(protoRoleType.getLabel());
```

Comment:
You can actually now call `conceptMgr.getType(Label.of(...)).asRoleType()` :) This looks a lot like what should already exist in ConceptManager

Reply:
`getType()` does not (and did not) exist on `ConceptManager`. Are you saying I should add it?

Reply:
Ah yeah you can add it like this (Sorry I forgot I had it in the Fetch PR):
```
  public Type getType(Label label) {
      TypeVertex vertex = graphMgr.schema().getType(label);
      if (vertex != null) {
          if (label.scope().isPresent()) return convertRoleType(vertex);
          else return convertThingType(vertex);
      } else return null;
  }
```
Then you can create a core Label structure from the message and use that